File: solved.ac - CLASS/CLASS 3/boj_5430.cpp
#include <bits/stdc++.h>
using namespace std;

typedef long long ll;
typedef pair<int, int> pii;
typedef pair<ll, ll> pll;

void solve()
{
	string p; cin >> p;
	int n; cin >> n;

	deque<int> dq;
	string str; cin >> str;
	string num = "";
	for(char c : str)
	{
		if(isdigit(c)) num += c;
		else
		{
			if(num != "") dq.push_back(stoi(num));
			num = "";
		}
	}

	bool isFront = true;
	for(char c : p)
	{
		if(c == 'R') isFront = !isFront;
		else
		{
			if(dq.empty()) 
			{
				cout << "error\n";
				return;
			}
			else
			{
				if(isFront) dq.pop_front();
				else dq.pop_back();
			}
		}
	}

	cout << "[";
	while(!dq.empty())
	{
		if(isFront)
		{
			cout << dq.front();
			dq.pop_front();
		}
		else
		{
			cout << dq.back();
			dq.pop_back();
		}
		
		if(dq.empty()) continue;
		cout << ",";
	}
	cout << "]\n";
}

int main()
{
	ios::sync_with_stdio(0);
	cin.tie(0), cout.tie(0);

	int T; cin >> T;
	while(T--) solve();
}

File: solved.ac - CLASS/CLASS 3/boj_5430.py
from collections import deque

def solve():
    p = input()
    n = int(input())
    str = input()[1:-1].split(',')

    if(str == ['']): X = deque()
    else: X = deque(str)
    
    isFront = True
    for c in p:
        if(c == 'R'): isFront = not isFront
        else:
            if(len(X) == 0):
                print("error")
                return
            else:
                if(isFront): X.popleft()
                else: X.pop()
    
    if(isFront == False): X.reverse()
    print('[', end='')
    print(','.join(X), end='')
    print(']')

T = int(input())
for t in range(T):
    solve()